Week2.java

```diff
@@ -0,0 +1,98 @@
+package Homework;
+
+
+/*******************WEEK2****************************/
+
+/*Problem: triplets that sum to zero. 
+  TALK:
+Assuming unique elements in the array.
+taking size of the array as input.
+
+ EXAMPLES:
+ array of size=0, output: no triplet 
+ array of size<3 ([1,2]), output: no triplet  
+size= 5  [1,2,3,-1,-2],    output: -2 -1 3
+size=6   [5,-1,-4,-2,6],   output: -4 -1 5
+                                   -4 -2 6
+ size=4  [1,2,3,4],        output: no triplet                                   
+
+BRUTE FORCE:
+considering all the triplets.
+complexity: O(n3)
+
+OPTIMIZED SOLUTION AS BELOW 
+*/
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+
+//IMPLEMENT
+
+public class Week2 {
+    
+	 public static void TakeInput(int[] array, int size) {
+	    	if(size==0) {
+	    		return;
+	    	}
+	    	System.out.print("do");
+	    	Scanner s=new Scanner(System.in);
+	    	for(int i=0;i<size;i++) {
+	    		array[i]=s.nextInt();
+	    	}
+	    }
+	public static void SumOfTriplets(int []input, ArrayList<Variable> solution) {
+    	int j,k;
+    	for(int i=0;i<input.length-2;i++) {
+    		j=i+1;
+    		k=input.length-1;
+    		while(j<k) {
+    			int sum=input[j]+input[k]+input[i];
+    			if(sum>0) {
+    			k--;	
+    			}
+    			else if(sum<0) {
+    				j++;
+    			}
+    			else {
+    				Variable ans=new Variable();
+    				ans.Add(input[i], input[j], input[k]);
+    				solution.add(ans);

```

Comment:
Overall this looks correct, but there is one thing that I noticed you are missing that would actually be really easy to fix since you created a new class. Currently, your code does not handle duplicate entries that are the same (which is also a requirement of the problem). For example:
Input: [ -1, -1, -1, 2] 
Expected output: [ [ -1, -1, 2] ]

But with your implementation, it would actually return the same expected output twice. 

One way to fix this easily is to implementing the Comparable interface on your Variable class and using a Set instead of an ArrayList. A set guarantees that elements in it are unique, and if you implement Comparable, you can use that to check if the Variables are equal and return that there. The Set will then call your compareTo method when adding and if it detects one that is equal in the set already, will not add it